### DISCUSSION TITLE
iox_add_executable and add_executable difference

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-iceoryx/iceoryx#1680] [Q&A] [answered]
Hello, sorry if this is the wrong place to ask, I am trying to get started and had a question regarding the CMake configuration...

I noticed that in the example projects a cmake directive "iox_add_executable" is used instead of the regular "add_executable"

What is the difference between the two?

If I wanted to add iceoryx to an existing CMake C++ project, which one should I use?

**Comment by elBoberido** [comment, marked as answer]
@AetherGlow `iox_add_executable` is used to hide some boilerplate code. It has some iceoryx specifics and therefore should not be used outside of iceoryx. To integrate iceoryx into your cmake project please have a look at https://github.com/eclipse-iceoryx/iceoryx-project-template/
